src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/FinalizeStep/hooks.tsx

```diff
@@ -0,0 +1,205 @@
+import React, { useEffect, useState } from 'react';
+import { BigNumber } from 'ethers';
+import { useLocation } from 'react-router-dom';
+import { ColonyActionType, StakerRewards, UserStakes } from '~gql';
+import { useAppContext, useColonyContext } from '~hooks';
+import { mapPayload } from '~utils/actions';
+import { formatText } from '~utils/intl';
+import Numeral from '~shared/Numeral';
+import { getBalanceForTokenAndDomain } from '~utils/tokens';
+import { MotionFinalizePayload } from '~redux/types/actions/motion';
+import { DescriptionListItem } from '../VotingStep/partials/DescriptionList/types';
+import { MotionAction } from '~types/motions';
+import { useUserTokenBalanceContext } from '~context';
+import { getTransactionHashFromPathName } from '~utils/urls';
+import { ClaimMotionRewardsPayload } from '~redux/sagas/motions/claimMotionRewards';
+import { RefetchAction } from '~common/ColonyActions/ActionDetailsPage/useGetColonyAction';
+
+export const useFinalizeStep = (actionData: MotionAction) => {
+  const {
+    motionData: { nativeMotionDomainId, motionId, gasEstimate },
+    type,
+    amount,
+    fromDomain,
+    tokenAddress,
+  } = actionData;
+  const { colony } = useColonyContext();
+  const { balances } = colony || {};
+  const { user } = useAppContext();
+
+  const domainBalance = getBalanceForTokenAndDomain(
+    balances,
+    tokenAddress ?? '',
+    Number(nativeMotionDomainId),
+  );
+
+  const requiresDomainFunds: boolean =
+    !!fromDomain &&
+    !!amount &&
+    type !== ColonyActionType.MintTokensMotion &&
+    type !== ColonyActionType.EmitDomainReputationPenaltyMotion &&
+    type !== ColonyActionType.EmitDomainReputationRewardMotion;
+
+  const isFinalizable =
+    !requiresDomainFunds ||
+    // Safe casting since if requiresDomainFunds is true, we know amount is a string
+    BigNumber.from(domainBalance ?? '0').gte(amount as string);
+
+  const transform = mapPayload(
+    () =>
```

Comment:
Can you just define the return type here instead of using casing?

Reply:
fixed